server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/controllers/CheckTokenController.java

```diff
@@ -0,0 +1,88 @@
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+
+@RestController
+@Slf4j
+@RequestMapping(value = "/check")
+public class CheckTokenController {
+
+    @Value("${security.oauth.clientId}")
+    private String oauthClientId;
+
+    @Value("${security.oauth.clientSecret}")
+    private String oauthClientSecret;
+
+    @Value("${security.oauth.checkTokenUrl}")
+    private String oauthCheckTokenUrl;
+
+    @Value("${security.oauth.authTokenUrl}")
+    private String authTokenUrl;
+
+    private static final Logger logger = LoggerFactory.getLogger(CheckTokenController.class);
+
+    @GetMapping(value = "/checkToken", headers = "Accept=*/*", produces = {"application/json", "text/xml"})
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = CollectionModel.class))),
+        @ApiResponse(responseCode = "400", description = "Bad Request", content = @Content(schema = @Schema(implementation = MessageListRsrc.class))),
+        @ApiResponse(responseCode = "401", description = "Unauthorized"),
+        @ApiResponse(responseCode = "403", description = "Forbidden"),
+        @ApiResponse(responseCode = "404", description = "Not Found"),
+        @ApiResponse(responseCode = "409", description = "Conflict"),
+        @ApiResponse(responseCode = "412", description = "Precondition Failed"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error", content = @Content(schema = @Schema(implementation = MessageListRsrc.class)))})
+    @Parameters({
+        @Parameter(name = HeaderConstants.VERSION_HEADER, description = HeaderConstants.VERSION_HEADER_DESCRIPTION, required = false, schema = @Schema(implementation = Integer.class), in = ParameterIn.HEADER),
+        @Parameter(name = HeaderConstants.IF_MATCH_HEADER, description = HeaderConstants.IF_MATCH_DESCRIPTION, required = true, schema = @Schema(implementation = String.class), in = ParameterIn.HEADER)})
+    public CheckedToken token(HttpServletRequest request, HttpServletResponse response) throws Oauth2ClientException, IOException {
+        logger.debug("<checkToken");
+
+        TokenServiceImpl tokenService;
+
+        tokenService = new TokenServiceImpl(
+                oauthClientId,
+                oauthClientSecret,
+                oauthCheckTokenUrl,
+                authTokenUrl);
+
+        CheckedToken result = null;
+        String authorizationHeader = request.getHeader("Authorization");
+        request.getSession().setAttribute("authToken", authorizationHeader);
+        try {
+            if (authorizationHeader == null) {
+                response.sendError(401);
+            } else {
+                result = tokenService.checkToken(authorizationHeader.replace("Bearer ", ""));
+            }
+        } catch (Oauth2ClientException | IOException t) {
+            response.sendError(500, t.getMessage());
```

Comment:
As a general rule, we shouldn't be sending a stack trace back to the client.  A friendlier message like "we were unable to process this request.  Please try again later.

Also, we should be logging these exceptions as well

Reply:
Both done